### DISCUSSION TITLE
Open from actual Livewire component

### DISCUSSION THREAD

**Discussion by MrMooky** [wire-elements/modal#159] [Q&A]
Is there any way to open a model with parameters from within a Livewire PHP component?

I tried a few variations of `$this->emit("openModal", "submission-status-confirm", json_encode(["submission_id" => $this->submission->id, "status_id" => $status]));`, but always ended up with:

> Unable to resolve dependency [Parameter #0 [ <required> $submission_id ]] in class ...

When I remove the dependency from the `mount()` it opens fine. What am I missing?

**Comment by PhiloNL** [comment]
Can you share a bit more code (component and blade)? Does the mount method looks like this?

```php
public function mount($submission_id, $status_id)
```

**Comment by dsbilling** [comment]
I am having the same issue.
This: `dd($this->emit('openModal', 'resource.update-create', json_encode(['resource_uuid' => $this->resource->uuid])));` returns:
```PHP
^ Livewire\Event {[#2411 ▼]()
  #name: "openModal"
  #params: array:2 [[▼]()
    0 => "resource.update-create"
    1 => "{"resource_uuid":"b21e9cf5-9ea5-4dcf-9db4-be2e193d4ab2"}"
  ]
  #up: null
  #self: null
  #component: null
}
```

And in the resource.update-create component I have tried dumping but it just shows up blank:
```PHP
public function mount($resource_uuid)
{
    dd($resource_uuid);
}
```

![image](https://user-images.githubusercontent.com/9788214/159567384-3cd12487-6d8c-42f4-b9f8-54b74468aec9.png)

**Reply by dsbilling** [reply]
For reference:
- https://github.com/P3D-Legacy/pokemon3d.net/blob/develop/app/Http/Livewire/Resource/ResourceForm.php
- https://github.com/P3D-Legacy/pokemon3d.net/blob/develop/app/Http/Livewire/Resource/UpdateCreate.php

**Comment by DanielSpravtsev** [comment]
Do not use json_encode([data]) inside Livewire component emit 

**Example**
`
$this->emit('openModal','app.modals.add-period', ["sim" => $this->sim->id]);
` 
This should work
